superset/assets/javascripts/explorev2/components/ChartContainer.jsx

```diff
@@ -57,7 +62,11 @@ class ChartContainer extends React.Component {
   componentWillReceiveProps(nextProps) {
     if (nextProps.chartStatus !== this.props.chartStatus
       || nextProps.height !== this.props.height) {
-      this.setState({ mockSlice: this.getMockedSliceObject(nextProps) });
+      const mockSlice = this.getMockedSliceObject(nextProps);
+      this.setState({
+        mockSlice,
+        viz: visMap[nextProps.viz_type](mockSlice),
```

Comment:
if the height is different you will re-create the object, which means issuing only a `viz.resize()` won't work because it depends on having had `viz.render()` executed on the same object, but here you're reseting that object.

Please test using an nvd3 viz and put console.log items identifying when render/resize are called and make sure they get called only when needed.